File: ClientSide/src/app/pages/students/models/studentDataModel.ts
import { StudentModel } from './studentModel';
import { ExtraSelectModel } from 'src/app/common/models/extraSelectModel';

export class StudentDataModel {

    public student: StudentModel;
    public faculties: Array<ExtraSelectModel>;

    constructor (
        student: StudentModel,
        faculties: Array<ExtraSelectModel>
    ) {
        this.student = student;
        this.faculties = faculties;
    }

}

File: ServerSide/SAMS.REST.API/DataModels/StudentDataModel.cs
using SAMS.BusinessLogic.Models.Common;
using SAMS.BusinessLogic.Models.Students;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SAMS.REST.API.DataModels
{
    public class StudentDataModel
    {
        public StudentModel Student { get; set; }
        public IQueryable<ExtraSelectModel> Faculties { get; set; }
    }
}
